addons/mail/static/src/core/common/store_service.js

```diff
@@ -393,31 +393,50 @@ export class Store extends BaseStore {
     }
 
     getMentionsFromText(
-        body,
+        { body, isHtmlBody },
         { mentionedChannels = [], mentionedPartners = [], specialMentions = [] } = {}
     ) {
         const validMentions = {};
-        validMentions.threads = mentionedChannels.filter((thread) => {
-            if (thread.parent_channel_id) {
-                return body.includes(
-                    `#${thread.parent_channel_id.displayName} > ${thread.displayName}`
-                );
-            }
-            return body.includes(`#${thread.displayName}`);
-        });
-        validMentions.partners = mentionedPartners.filter((partner) =>
-            body.includes(`@${partner.name}`)
-        );
-        validMentions.specialMentions = this.specialMentions
-            .filter((special) => body.includes(`@${special.label}`))
-            .map((special) => special.label);
+        if (isHtmlBody) {
```

Comment:
This should be a simple check on whether body is markup or not, we don't want to pass an extra flag around.